src/app/games/hive/hive.component.html

```diff
@@ -0,0 +1,73 @@
+<svg xmlns="http://www.w3.org/2000/svg"
+     width="100%" height="100%"
+     [attr.viewBox]="viewBox"
+     preserveAspectRatio="xMidYMid meet">
+    <g *ngFor="let groundInfo of ground">
+        <polygon id="space_{{ groundInfo.x }}_{{ groundInfo.y }}"
+                 [attr.points]="getHexaPointsAtXY(groundInfo.x, groundInfo.y)"
+                 class="base mid-stroke"
+                 [ngClass]="groundInfo.content.spaceClasses"
+                 (click)="selectSpace(groundInfo.x, groundInfo.y)"/>
+    </g>
+    <g *ngFor="let groundInfo of ground">
+        <polygon *ngIf="groundInfo.content.strokeClasses.length > 0"
+                 id="stroke_{{ groundInfo.x }}_{{ groundInfo.y }}"
+                 [attr.points]="getHexaPointsAtXY(groundInfo.x, groundInfo.y)"
+                 class="base no-fill small-stroke"
+                 [ngClass]="groundInfo.content.strokeClasses"
+                 (click)="selectSpace(groundInfo.x, groundInfo.y)"/>
+    </g>
+    <g *ngFor="let level of pieces; index as z">
+        <g *ngFor="let pieceInfo of level">
+            <g id="piece_{{ pieceInfo.x }}_{{ pieceInfo.y }}"
+               app-hive-piece
+               [piece]="pieceInfo.content.piece"
+               [x]="pieceInfo.x"
+               [y]="pieceInfo.y"
+               [pieceHeight]="PIECE_HEIGHT"
+               [hexaLayout]="hexaLayout"
+               [level]="z"
+               (click)="selectStack(pieceInfo.x, pieceInfo.y)"/>
+        </g>
+        <g *ngFor="let pieceInfo of level">
+            <polygon *ngIf="pieceInfo.content.strokeClasses.length > 0"
+                     id="stroke_{{ pieceInfo.x }}_{{ pieceInfo.y }}"
+                     [attr.transform]="'translate(0 -' + (z+1) * PIECE_HEIGHT + ')'"
+                     [attr.points]="getHexaPointsAtXY(pieceInfo.x, pieceInfo.y)"
+                     class="base no-fill small-stroke"
+                     [ngClass]="pieceInfo.content.strokeClasses"
+                     (click)="selectStack(pieceInfo.x, pieceInfo.y)"/>
+        </g>
+    </g>
+    <g *ngFor="let stack of remainingPieces">
+        <g *ngFor="let piece of stack.pieces; index as z">
```

Comment:
let remainingStack of remainingStacks :p ? (et remainingStack_"

Reply:
renommé piece en stack mais pas l'id html (car l'élément correspond à une seule pièce)